test/end-to-end/test_pkg_bulkupload.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Basic set of tests for the hab pkg bulkupload command
+#
+# Assumptions:
+# 1. ACCEPTANCE_HAB_AUTH_TOKEN or HAB_AUTH_TOKEN Environment variables are set and valid
+# 2. ${CACHE_DIR} can be set to a writable location on the filesystem
+# 3. non zero exit code from each command implies success
+
+# TODO: Future improvement would be to create a clean room Builder or origin.
+# Doing so would allow for more accurate validation of the bulkupload by interrogating
+# the Builder. Of course this introduces significant setup time cost..
+
+set -euo pipefail
+
+export HAB_NOCOLORING=true
+export HAB_NONINTERACTIVE=true
+export HAB_AUTH_TOKEN=${ACCEPTANCE_HAB_AUTH_TOKEN:-${HAB_AUTH_TOKEN}}
+export HAB_ORIGIN="testbulkupload"
+export HAB_BLDR_URL=${HAB_BLDR_URL:-https://bldr.acceptance.habitat.sh}
+
+HAB=${HAB_TEST_CMD:-hab}
+CACHE_DIR="test-cache"
+FIXTURES_DIR="fixtures/bulkupload"
+PKG_A_TAR="${FIXTURES_DIR}/testbulkupload-testpkg1-0.1.0-20191024190939.tar"
+PKG_B_TAR="${FIXTURES_DIR}/testbulkupload-testpkg2-0.1.0-20191024191005.tar"
+PKG_A_HART="${CACHE_DIR}/artifacts/testbulkupload-testpkg1-0.1.0-20191024190939-x86_64-linux.hart"
+PKG_B_HART="${CACHE_DIR}/artifacts/testbulkupload-testpkg2-0.1.0-20191024191005-x86_64-linux.hart"
+
+echo
+echo "--- Testing with command ${HAB}, using cache dir ${CACHE_DIR}"
+echo
+
+before_upload() {
```

Comment:
I thing these are fine, but it seems like we're very quickly approaching a point where we need to bring in a more formal framework of some kind. For instance, all this seems like it wants to be `bats`, and I've actually got some current work where getting a test environment for `bats` set up (with all the helper libraries) doesn't actually suck. `aruba` is another nice library in the Ruby world. And we've also discussed some kind of Powershell path to make cross-platform testing a bit easier.

I think these tests that you're writing are a bit different from other ones we've got, in that they're more about exercising CLI activity (others are about Supervisor functionality, which bring with them other constraints). That's a distinction that may be worth considering as we move forward.

Reply:
Also, this is just an observation / food for thought... I'm not asking for a rewrite in `bats` (or anything else!) for this PR :sweat_smile:

Reply:
I'm +1 for merging this as is, but I think we should plan a spike near term to see if it's feasible to write a CLI test in powershell and make it work cross platform. If so, we should make that our standard going forward, and rewrite tests as a tech debt item.